chained/water_impl.go

```diff
@@ -39,40 +38,44 @@ func newWaterImpl(addr string, pc *config.ProxyConfig, reportDialCore reportDial
 		}
 	}
 
+	ctx := context.Background()
 	wasmAvailableAt := ptSetting(pc, "water_available_at")
+	transport := ptSetting(pc, "water_transport")
 	if wasm == nil && wasmAvailableAt != "" {
-		urls := strings.Split(wasmAvailableAt, ",")
-		b := new(bytes.Buffer)
-		cli := httpClient
-		if cli == nil {
-			downloadTimeout := ptSetting(pc, "water_download_timeout")
-			timeout := 10 * time.Minute
-			if downloadTimeout != "" {
-				var err error
-				timeout, err = time.ParseDuration(downloadTimeout)
-				if err != nil {
-					return nil, fmt.Errorf("failed to parse download timeout: %w", err)
-				}
-			}
-
-			cli = proxied.ChainedThenDirectThenFrontedClient(timeout, "")
+		waterDir := filepath.Join(configDir, "water")
+		vc, err := NewVersionControl(waterDir)
```

Comment:
Could you instead have `newWaterImpl` just accept a path to a version control directory?  `newWaterImpl` doesn't need to know about the existence of a config directory or manage anything in the directory containing the version control folder.

Reply:
In this case, how should we set this? Which package or function should be responsible for setting the water dir? `chained/proxies.go` refers to `newWaterImpl` for initializing, should it be responsible for setting the waterDir?

Reply:
Answered in another comment [here](https://github.com/getlantern/flashlight/pull/1427/files#r1817085389).